site/src/components/Resources/AgentRow.tsx

```diff
@@ -396,6 +380,51 @@ export const AgentRow: FC<AgentRowProps> = ({
   );
 };
 
+const useAgentLogs = (
+  agentId: string,
+  { enabled, initialData }: { enabled: boolean; initialData?: LineWithID[] },
+) => {
+  const [logs, setLogs] = useState<LineWithID[] | undefined>(initialData);
```

Comment:
Is there a benefit to having this be potentially `undefined`, or would it be possible to define this only as `LineWithID[]`, and accept empty arrays?

Reply:
Yes, when it is undefined we know it is not loaded yet

Reply:
Okay, that makes sense